### DISCUSSION TITLE
Memory keyword

### DISCUSSION THREAD

**Discussion by MuhammadAmmar24** [nehal2210/Blockchain-Engineer-Course#12] [Q&A] [answered]
Why we always write **string** **memory** on declaration of a string? Is it right to declare uint with memory key word?

**Comment by imtiaz70** [comment, marked as answer]
memory is a keyword used to store data for the execution of a contract , and writing a memory keyword with uint is not the best practice because a unit does not require large space(memory) to store data, string stores large data so it requires a memory keyword.
